app/code/Magento/Store/Model/App/EnvironmentEmulation.php

```diff
@@ -0,0 +1,241 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+/**
```

Comment:
Please move the description to the class definition and add some details it if possible